<!--
  Multiradar (1642) full sensorsuite launch file

  Launshes:
    - 1x 1642 radar
    - 1x LORD microstrain IMU
    - 1x Intel Realsense d435
 -->

<launch>

  <arg name="rviz" default="false"/>
  <arg name="radar" default="true"/>
  <arg name="camera" default="true"/>
  <arg name="imu" default="true"/>

  <arg name="radar_name_0" default="radar_fwd" doc="forward facing radar"/>
  <arg name="realsense_name_0" default="camera" doc="forward facing realsense"/>

  <arg name="vicon_server" default="tracker"/>
  <arg name="tracker_names" default="radar_rig"/>

  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_node" output="screen">
    <!-- param name="server" value="192.168.50.10"/ -->
    <param name="server" value="$(arg vicon_server)"/>
    <rosparam param="tracker_name" subst_value="True">$(arg tracker_names)</rosparam>
    <param name="refresh_tracker_frequency" value="10"/>
    <param name="use_server_time" value="true"/>
  </node>

  <include if="$(arg radar)" file="$(find ti_mmwave_rospkg)/launch/ti_mmwave_sensor.launch">
    <!-- set arg values for radar0 -->
    <arg name="device" value="1843"/>
    <arg name="config" value="3d_no_grp_10Hz"/>
    <arg name="sdk" value="." doc="{AWR1642} [02_01_00_04], {IWR1642, AWR1443, IWR1443} [01_02_00_05], {AWR1843} [03_01_01_02]"/>
    <arg name="command_port" value="/dev/ttyACM0"/>
    <arg name="data_port" value="/dev/ttyACM1"/>
  </include>

  <!-- launch realsense camera node -->
  <include if="$(arg camera)" file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="enable_color" value="false"/>
    <arg name="enable_gyro" value="false"/>
    <arg name="enable_accel" value="false"/>
    <arg name="enable_sync" value="true"/>
  </include>

  <!-- launch LORD Microstrain GX5-15 IMU -->
  <include if="$(arg imu)" file="$(find microstrain_driver)/launch/3dm_gx5_15.launch">
    <arg name="port" value="/dev/serial/by-id/usb-Lord_Microstrain_Lord_Inertial_Sensor_0000__6254.75760-if00"/>
    <arg name="frequency" value="200"/>
    <!-- remap from="imu_raw" to="imu/data_raw"-->
  </include>

  <!-- launch Rig 2 static transforms -->
  <include file="$(find static_tfs)/launch/rig2_transforms.launch">
    <arg name="radar" value="$(arg radar)"/>
    <arg name="camera" value="$(arg camera)"/>
    <arg name="imu" value="$(arg imu)"/>
  </include>

  <!-- Launch Rviz with pre-defined configuration to view mmWave sensor detected object data (color by intensity) -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(find ti_mmwave_rospkg)/rviz/radar_det_obj_color_by_elev.rviz"/>

  <!-- transform for base_radar_link to map (for RVIZ) -->
  <node if="$(arg radar)" pkg="tf" type="static_transform_publisher" name="radar_map_tf" args="0 0 0 0 0 0 base_radar_link map 10"/>

</launch>
